### PR TITLE
Process::Status#to_i: return raw waitpid(2) status

### PR DESCRIPTION
**Pull request by rafaelfranca**
Shopify#1

### CODE REVIEW

File: src/main/ruby/truffleruby/core/process.rb [hunk 1]
```diff
@@ -729,19 +731,20 @@ class Status
     attr_reader :termsig
     attr_reader :stopsig
 
-    def initialize(pid=nil, status=nil, termsig=nil, stopsig=nil)
+    def initialize(pid=nil, status=nil, termsig=nil, stopsig=nil, raw_status=nil)
       @pid = pid
       @status = status
```
Comment (eregon) [inline src/main/ruby/truffleruby/core/process.rb]: I think we should rename this to `@exitstatus` to avoid confusion and mirror the C macro names.

**Comment by rafaelfranca** [comment]
I run all the test in my machine and saw no failures with my last commit.

File: src/main/ruby/truffleruby/core/process.rb [hunk 2]
```diff
@@ -729,40 +731,41 @@ class Status
...
+      @raw_status >> num
     end
 
     def coredump?
       false
     end
 
     def exited?
-      @status != nil
+      @raw_status != nil
```
Comment (eregon) [inline src/main/ruby/truffleruby/core/process.rb]: This one should be `@exitstatus`

File: src/main/ruby/truffleruby/core/process.rb [hunk 3]
```diff
@@ -779,14 +782,14 @@ def stopped?
 
     def success?
       if exited?
-        @status == 0
+        @raw_status == 0
```
Comment (eregon) [inline src/main/ruby/truffleruby/core/process.rb]: This one should be `@exitstatus`

File: src/main/ruby/truffleruby/core/process.rb (hunk 2 above)
Comment (eregon) [reply src/main/ruby/truffleruby/core/process.rb]: I'll fix those two :)

Comment (rafaelfranca) [reply src/main/ruby/truffleruby/core/process.rb]: Thanks. We will need test coverage on those as well.

Comment (eregon) [reply src/main/ruby/truffleruby/core/process.rb]: Good point, actually there are already specs but one of the `exited?` specs was tagged. That one fails with `@raw_status`.
`@raw_status` seems to work in `success?` but I think it's just luck of the native encoding for exited.

Comment (eregon) [reply src/main/ruby/truffleruby/core/process.rb]: Together with #1790 we pass all Process::Status specs.

**Pull request merged**